aws/logs_monitoring/lambda_function.py

```diff
@@ -983,7 +983,17 @@ def cwevent_handler(event, metadata):
 
 
 # Handle Sns events
-def sns_handler(event, metadata):
+def sns_handler(event, context, metadata):
+    # test if sns payload contains an s3 event
+    try:
+        event = json.loads(event['Records'][0]['Sns']['Message'])
```

Comment:
I think there is a bug here. If the event successfully parses as json, but fails any of the below checks , it will overwrite the original event object, and the rest of the sns_handler function will  be operating on the wrong object. We are currently working on building an integration testing suite to catch regressions for things like this.

Reply:
Fixed. I moved the `data = event` back to the top of the function, and use `data` in the rest of the function.

Reply:
My other suggestions would be to move this logic into the `for ev in data["Records"]:`, other only the first event in the list is being processed. Also, it might make sense to reuse the parse method so any other event types submitted via SNS can be processed. eg

```python
for ev in data["Records"]:
   try:
      nested_event = json.loads(ev['Sns']['Message'])
      if isinstance(event, dict) and "Records" in event and hasattr(event["Records"], '__len__'):
         nested_evs = parse(event, context)
         for ne in nested_evs:
             yield ne
         return
   except json.decoder.JSONDecodeError:
      pass
   yield ev
```